src/lib/validator.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Values to be validated
+ */
+export interface IValidationValue {
+  value: undefined | null | string;
+  error: string;
+}
+
+/**
+ * Returns true of val is undefined, null or empty string.
+ *
+ * @param val Value to inspect
+ */
+export const hasValue = (val: undefined | null | string): boolean => {
```

Comment:
we could probably get extra use out of this function in other places if we convert this to a type-guard, something like:

```ts
type AtomicValue =
  | bigint
  | boolean
  | null
  | number
  | object
  | string
  | symbol
  | undefined;

const hasValue = (val: unknown): val is NonNullable<AtomicValue> => {
  return val !== undefined && val !== null && val !== "";
};
```